### PR TITLE
doc: net: Update network overview chapter

### PR DESCRIPTION
**Pull request by jukkar**
Fixing typos and missing features in network overview
chapter.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

### CODE REVIEW

**Review by nashif — APPROVED** [review]

**Comment by jukkar** [comment]
This PR depends on #13679 so I will re-submit this one after that PR is merged.

File: doc/guides/networking/overview.rst [hunk 1]
```diff
@@ -101,6 +100,11 @@ can be disabled if not needed.
 * **Time Sensitive Networking.** The gPTP (generalized Precision Time Protocol)
   is supported. See :ref:`gptp_interface` for more details.
 
+* **Network shell.** The network shell provides helpers for figuring out
+  network status, enabling/disabling features, and issuing commands like ping
+  or DNS resolving. The net-shell is useful when developing network software.
+  See :ref:`net_shell` for more details.
```
Comment (dbkinder) [inline doc/guides/networking/overview.rst]: It looks like the doc label for this is ``:ref:`shell_label` ``  (which would link to https://docs.zephyrproject.org/latest/reference/shell/index.html)

**Review by dbkinder — CHANGES_REQUESTED** [review]

File: doc/guides/networking/overview.rst (hunk 1 above)
Comment (jukkar) [reply doc/guides/networking/overview.rst]: Actually I was waiting #13679 that will introduce the net_shell label.

**Comment by jukkar** [comment]
Rebased so that net_shell reference is now found.

**Review by dbkinder — APPROVED** [review]

**Pull request merged**